### ISSUE TITLE
Custom Data fetching hook built on top of useQuery fails due to undefined response

### ISSUE THREAD

**Issue by goleary** [blitz-js/blitz#1379]
### What is the problem?

I'm trying to define a custom data hook that is built on top of Blitz's `useQuery` hook.

Here is my hook code:

```js
import { Entry } from "@prisma/client"
import getEntries from "app/entries/queries/getEntries"
import { useQuery } from "blitz"
import { useCurrentUser } from "./useCurrentUser"

type Aggregation = "week" | "month" | "year"

export const useEntries = (aggregation?: Aggregation) => {
  const user = useCurrentUser()

  const [{ entries }] = useQuery(
    getEntries,
    {
      where: { userId: user?.id },
      orderBy: { createdAt: "desc" },
    },
    { enabled: user }
  )
  // do stuff
  return { entries }
}
```

When I call `useQuery` with `getEntries` (generated by Blitz) it returns an undefined response.  The type returned by useQuery in this instance is a response that always has an `entries` property, but in practice that is not what I'm seeing happen.

Error:

```
Uncaught TypeError: Cannot read property 'entries' of undefined
```

I'm a little perplexed why this code (generated by Blitz) works:

```js
 const [{ entries }] = useQuery(getEntries, {
    orderBy: { id: "asc" },
  })

  return (
    <div>
      <ul>
        {entries.map((entry) => (
          <li key={entry.id}>
            <Link href="/entries/[entryId]" as={`/entries/${entry.id}`}>
              <a>{entry.text}</a>
            </Link>
          </li>
        ))}
      </ul>
```

My best guess is that `useQuery` is using `Suspense` in some way that I don't understand (I'm not familiar with it), but that when it's called from another hook `Suspense` doesn't prevent the subsequent code from being executed and I get a `undefined` reference error.

Even when I pass `{suspense: false}` to `useQuery` I'm still getting this error.

### Steps to Reproduce

You can find my code here: https://github.com/goleary/olad/blob/use-query-error/app/hooks/useEntries.ts

### Versions

```
Windows 10 | win32-x64 | Node: v12.16.3

blitz: 0.24.3 (global)
blitz: 0.24.3 (local)

  Package manager: yarn
  System:
    OS: Windows 10 10.0.19041
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 15.47 GB / 31.86 GB
  Binaries:
    Node: 12.16.3 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - ~\.yarn\bin\yarn.CMD
    npm: 6.14.4 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  npmPackages:
    @prisma/cli: 2.9.0 => 2.9.0
    @prisma/client: 2.9.0 => 2.9.0
    blitz: 0.24.3 => 0.24.3
    react: 0.0.0-experimental-7f28234f8 => 0.0.0-experimental-7f28234f8
    react-dom: 0.0.0-experimental-7f28234f8 => 0.0.0-experimental-7f28234f8
    typescript: 4.0.3 => 4.0.3
```

**Comment by flybayer** [comment]
Hey @goleary! That definitely is weird. I can't think of why that wouldn't work. We don't mess with the hooks/components during the build, so what you see is what you _should_ get :) 

Someone just needs to dig in and do some debugging on this.

**Comment by mtford90** [comment]
Experiencing this also.

If usage of `useQuery` is extracted to another hook, then suspense seems to be disabled when using this hook even if `suspense: true`. Pretty weird.

**Comment by mtford90** [comment]
Ah - is it because `useSession` does not suspend, and therefore the `useQuery` call waits for `enabled` to be `true` and therefore never suspends either? I guess this isn't a good workaround after all.

**Comment by flybayer** [comment]
Ok bummer. Will try to get this fixed asap.

**Comment by flybayer** [comment]
Fixed in #1888
